pkg/build/testdata/buildkit.Dockerfile

```diff
@@ -1,5 +1,5 @@
 # syntax=docker/dockerfile-upstream:1.4.0
-FROM debian:stretch-slim
+FROM --platform=linux/amd64 debian:stretch-slim
```

Comment:
I think that there was a bit of a misunderstanding from my previous feedback. I was only asking that you update this file so that when someone does not have a custom dockerfile specified, the template used during porter build has a platform set.

You should not modify the contents of the custom dockerfile when set (which is what I think you were doing with buildFROMSection?)

Reply:
> Can you update that file and then check that we have a test that will validate the platform is set both for people using a custom template and those who aren't?

Sorry this was unclear! What I meant was that:

* When the custom dockerfile is generated by porter during `porter create`, the template.Dockerfile should have a platform set.
* When porter build is run and a custom dockerfile is not set in porter.yaml (by the template field), the template used during porter build should include the platform
* When the user has a custom dockerfile specified in their porter.yaml, we should not attempt to insert a platform. This allows them to customize the platform themselves, or do whatever special stuff they require without Porter's interference.

Reply:
Thank you for the clarification! I updated the code to reflect it